### PR TITLE
Test: updated cloudfront Resource generation `runServerless` unit test

### PR DESCRIPTION
**Pull request by jindalAnuj**
Addresses: #8528

### CODE REVIEW

**Review by medikoo — COMMENTED** [review]
@jindalAnuj once it's ready be sure to switch it to regular PR and mentioning me. Thanks!

**Comment by jindalAnuj** [comment]
@medikoo for sure

**Comment by jindalAnuj** [comment]
@medikoo i have completed all the `resource generation` test.
But there is a test case to test lambda versions which i am not able to get.
https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L283-L315

the thing is actually `versionFunctions` is not doing anything for me can you point me to some done where i can get the implementation for the same to write the test case

**Comment by medikoo** [comment]
> the thing is actually versionFunctions is not doing anything for me can you point me to some done where i can get the 

`versionFunctions` is internal property which should not be referenced anyway when we test with help of `runServerless`.

In new test we should simply test, that all functions which have `cloudFront` event, have versioning turned on, and that should be confirmed by inspecting the CF stack (in a similar manner when we test if the function is versioned when it's explicitly set to be versioned: https://github.com/serverless/serverless/blob/b5509cfdfe2b978fe94820f5080b32e95d0f1e36/test/unit/lib/plugins/aws/package/compile/functions.test.js#L1341)

**Comment by jindalAnuj** [comment]
@medikoo moved this pr to ready for review

**Comment by jindalAnuj** [comment]
@medikoo somehow codecov is giving falsy result

File: test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js [hunk 1]
```diff
@@ -5,8 +5,9 @@ const AwsProvider = require('../../../../../../../../lib/plugins/aws/provider');
 const AwsCompileCloudFrontEvents = require('../../../../../../../../lib/plugins/aws/package/compile/events/cloud-front');
 const Serverless = require('../../../../../../../../lib/serverless');
 const runServerless = require('../../../../../../../utils/run-serverless');
+const sandbox = require('sinon-chai');
 
-chai.use(require('sinon-chai'));
+chai.use(sandbox);
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js:10]: I think this change is not necessary?

File: test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js [hunk 2]
```diff
@@ -1620,62 +955,163 @@ describe('test/unit/lib/plugins/aws/package/compile/events/cloudFront.test.js',
...
+      ]);
+
+      expect(cfDistribution.Properties.DistributionConfig.Origins.length).to.greaterThanOrEqual(2);
     });
 
-    it.skip('TODO: should create different origins for the same domains with the same path but different protocols', () => {
+    it('should create different origins for the same domains with the same path but different protocols', () => {
       // Replaces
-      // https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L665-L735
+      // -- https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L665-L735
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js:1022]: I think we should remove those comments

File: test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js [hunk 3]
```diff
@@ -1620,62 +955,163 @@ describe('test/unit/lib/plugins/aws/package/compile/events/cloudFront.test.js',
...
+      ]);
     });
 
-    it.skip('TODO: should support origin customization', () => {
+    it('should support origin customization', () => {
       // Replaces
-      // https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L855-L901
-      // https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L903-L986
+      // -- https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L855-L901
+      // -- https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L903-L986
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js:1072]: Same here

**Review by medikoo — COMMENTED** [review]
@jindalAnuj that looks really good.

Please see my comments. 

Concerning codecov, by checking it's report, it complains that we no longer test one code path related to bevaviors (maybe we can improve that?)
Also it complains that `provider.getStackResources()` is no longer evaluated. I'm not sure exactly what old test trigger that, but as I investigate the code, there's no single reference to that method, it looks as dead code

**Comment by jindalAnuj** [comment]
@medikoo this pr is ready for review resolved all comments

**Review by medikoo — APPROVED** [review]
That's outstanding work @jindalAnuj, Thank you!

**Pull request merged**